sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -136,10 +137,52 @@ object DecimalType extends AbstractDataType {
     case DoubleType => DoubleDecimal
   }
 
+  private[sql] def forLiteral(literal: Literal): DecimalType = literal.value match {
```

Comment:
`fromLiteral`?

Reply:
since we have `forType` I used `forLiteral` to be coherent on the naming

Reply:
but we also have `fromDecimal`...